### PR TITLE
Inclusion of quantum excitation, code adaptation and bug fix.

### PR DESCRIPTION
**Pull request by xresende**
Check the PR https://github.com/lnls-fac/trackcpp/pull/53 for details about the quantum excitation implementation.

**Changes made:**
- The radiation_on property was changed to accept strings or integers corresponding to different radiative states (See #https://github.com/lnls-fac/trackcpp/pull/53);
- There were added functions to control the distribution and the seed of the pseudo-random number generator which is used to apply the quantum kicks;
- 6D find orbit was adapted to temporarily turn off quantum excitation before computing the orbit.
- Refactorization of some tests in the tests module. They were used to check if the addition of quantum excitation in the code broke something.
- Corrects a bug in lattice module: find_indices can't search for a given `polynom_b` or `polynom_a` because the `_isequal` excepts only a TypeError when checking if elements of an array-like are iterables, however, numpy.float64 returns an IndexError in this situation, which breaks the algorithm. I fixed it by putting the two errors as exceptions.

### CODE REVIEW

**Review by jquentino — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Review by murilobalves — APPROVED** [review]

**Review by Gabrielrezende-asc — APPROVED** [review]

**Review by fernandohds564 — APPROVED** [review]

File: pyaccel/tracking.py [hunk 1]
```diff
@@ -165,13 +165,16 @@ def generate_bunch(
...
-    accelerator.radiation_on = False
+    accelerator.radiation_on = 0
 
 
 @_interactive
-def set_6d_tracking(accelerator):
+def set_6d_tracking(accelerator, rad_full=False):
     accelerator.cavity_on = True
-    accelerator.radiation_on = True
+    if rad_full:
```
Comment (fernandohds564) [inline pyaccel/tracking.py:174]: maybe
```python3
accelerator.radiation_on = 2 if rad_full else 1
```
is more pythonic.

File: pyaccel/tracking.py [hunk 2]
```diff
@@ -607,6 +611,11 @@ def find_orbit6(accelerator, indices=None, fixed_point_guess=None):
     """
     indices = _process_indices(accelerator, indices)
 
+    # The orbit can't be found when quantum excitation is on.
+    if accelerator.radiation_on == 2:
+        rad_stt = accelerator.radiation_on
```
Comment (fernandohds564) [inline pyaccel/tracking.py:616]: you should always save the state. Put this line outside the if statement.

File: pyaccel/tracking.py [hunk 3]
```diff
@@ -616,6 +625,9 @@ def find_orbit6(accelerator, indices=None, fixed_point_guess=None):
 
     ret = _trackcpp.track_findorbit6(
         accelerator.trackcpp_acc, _closed_orbit, fixed_point_guess)
+
+    accelerator.radiation_on = rad_stt
```
Comment (fernandohds564) [inline pyaccel/tracking.py:629]: This is why line 616 is a bug.

**Review by fernandohds564 — CHANGES_REQUESTED** [review]
@jquentino, I think it is better to set the `radiation_on` property along the code with its string value, or at least put comments with the meaning of the numbers.

File: pyaccel/tracking.py (hunk 1 above)
Comment (jquentino) [reply pyaccel/tracking.py]: Done, thanks!

File: pyaccel/tracking.py (hunk 2 above)
Comment (jquentino) [reply pyaccel/tracking.py]: Done. I also put an observation in the docstring explaining that this function temporarily sets radiation_on to 'damping'. Thanks!

File: pyaccel/tracking.py (hunk 3 above)
Comment (jquentino) [reply pyaccel/tracking.py:628]: Fixed, thanks!

**Comment by jquentino** [comment]
> @jquentino, I think it is better to set the `radiation_on` property along the code with its string value, or at least put comments with the meaning of the numbers.

Ok, I did it. Thanks for the suggestion!

File: pyaccel/utils.py [hunk 4]
```diff
@@ -2,8 +2,6 @@
 
 import warnings
 import numpy as _numpy
-
-
```
Comment (fernandohds564) [inline pyaccel/utils.py:6]: leave one empty between builtin modules/packages imports and external modules/packages imports  and one empty line between external modules/packages imports and our own modules imports:

```python3
import warnings

import numpy as _numpy
 
import trackcpp as _trackcpp
```

Comment (jquentino) [reply pyaccel/utils.py:6]: Fixed, thanks!

**Review by fernandohds564 — APPROVED** [review]

**Pull request merged**